packages/cdktf-cli/lib/util.ts

```diff
@@ -91,4 +92,38 @@ export async function readCDKTFVersion(outputDir: string): Promise<string> {
 export function downcaseFirst(str: string): string {
   if (str === '') { return str; }
   return `${str[0].toLocaleLowerCase()}${str.slice(1)}`;
+}
+
+export class HttpError extends Error {
+  constructor(message?: string, public statusCode?: number) {
+    super(message); // 'Error' breaks prototype chain here
+    Object.setPrototypeOf(this, new.target.prototype); // restore prototype chain
+    // see: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-2.html#support-for-newtarget
+  }
+}
+
+export async function downloadFile(url: string, targetFilename: string): Promise<void> {
+  const client = url.startsWith('http://') ? http : https;
```

Comment:
I think it would be good to support file URI as well, so something like `file:///foo/bar/archive.zip`

Reply:
I created a follow up issue for this enhancement #653